v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/writer/DeadLetterQueue.java

```diff
@@ -83,11 +85,14 @@ public PTransform<PCollection<String>, PDone> getDlqTransform() {
   }
 
   private DeadLetterQueue(
-      String dlqDirectory, Ddl ddl, String shardIdColumn, SQLDialect sqlDialect) {
+      String dlqDirectory,
+      Ddl ddl,
+      Map<String, String> srcTableToShardIdColumnMap,
+      SQLDialect sqlDialect) {
     this.dlqDirectory = dlqDirectory;
     this.dlqTransform = createDLQTransform(dlqDirectory);
     this.ddl = ddl;
-    this.shardIdColumn = shardIdColumn;
+    this.srcTableToShardIdColumnMap = srcTableToShardIdColumnMap;
```

Comment:
nit: Should we create this new map and pass here rather than sticking to our common struct like `schemaMapper`  being pass around everywhere, and extend the mapper to support both source and spanner table names?

Reply:
The stuff passed to the transform needs to be serializable, so a complex object may not work.